demos/common/processing/include/async_pipeline.h

```diff
@@ -0,0 +1,136 @@
+/*
```

Comment:
The headers should be in a `processing` subdirectory, so that when they're included from the demos, the includes look like `#include <processing/async_pipe.h>`.

Reply:
why ? This is separate library, its include path is written in its own CMakeLists.txt. Let's not overcomplicate headers including process. BTW, currently all files are placed according to default Linux style - headers and cpp are in different directories.

Reply:
> why ? This is separate library

You immediately answered yourself. :-) If it's a separate library, its headers should be located in a subdirectory corresponding to the library name, so that they don't conflict and can be distinguished from headers from other libraries.

Reply:
that's really weird directories structure. We should either change cmake scripts or this rule. You suggest to create this structure:
common/processing/include/processing
Doesn't it look weird ? Most porjects keep headers inside include directory without any additions.
Also, could you provide any guide/code style that claims that we should include file with library name prefix ? I'd prefer to use it without that prefix - it's more convinient

Reply:
done